File: apps/salon/lib/models/message.dart
class Message {
  const Message({
    required this.id,
    required this.conversationId,
    required this.senderType,
    required this.senderName,
    required this.body,
    required this.createdAt,
  });

  final String id;
  final String conversationId;
  final String senderType;
  final String senderName;
  final String body;
  final DateTime createdAt;

  bool get isSalon => senderType.toLowerCase() == 'salon';

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id'] as String,
      conversationId: json['conversationId'] as String,
      senderType: json['senderType'] as String? ?? 'salon',
      senderName: json['senderName'] as String? ?? '',
      body: json['body'] as String? ?? '',
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}

File: services/api/Models/Message.cs
namespace ClientFlow.Api.Models;

public class Message
{
    public Guid Id { get; set; }
    public Guid ConversationId { get; set; }
    public Conversation? Conversation { get; set; }
    public string SenderType { get; set; } = "salon"; // salon or client
    public string SenderName { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}
